### PR TITLE
Add TemperatureGradient and deprecate LapseRate

### PR DESCRIPTION
**Pull request by dglozano**
I have added a new Quantity called TemperatureGradient and deprecated LapseRate, as agreed in #988 

In addition, I did a small change in the CodeGen project to place the `[Obsolete]` attribute below the XML documentation, to avoid getting some warnings. 

This is how it was generated before my change
image

I have a small issue though: after cloning the repo, and before doing any change at all, I run the CodeGen and I see tons of changes being made. It seems that much of the code was alphabetically ordered before, and the CodeGen by default in my machine overrides that (check https://github.com/angularsen/UnitsNet/commit/a9235843b197913f1b5a62b1476c36d03ce047e3)

Any suggestion on how to solve this ?

### CODE REVIEW

**Comment by angularsen** [comment]
> Any suggestion on how to solve this ?

Huh, that is super weird. Seems you are on MacOS or Linux maybe, I am on Windows. Could it be anything there?

It should be alphabetically sorted. Could you debug the codegen console app and maybe see why it produces a different ordering? Maybe there is some subtle difference on LINQ not preserving the ordering or something. I would expect adding a OrderBy() at the right place should fix this.

**Comment by angularsen** [comment]
> In addition, I did a small change in the CodeGen project to place the [Obsolete] attribute below the XML documentation, to avoid getting some warnings.

Sounds good to me 👍

**Comment by dglozano** [comment]
That's right, I am using Intellij on a Mac. I will debug the CodeGen later today and try to sort it out 😀

**Comment by dglozano** [comment]
It seems the culprit of the sorting issue was the `Directory.GetFiles` method  that's used to read the quantity json files.
It doesn't guarantee the order of the files. It seems that the result in Windows is always sorted alphabetically, but that wasn't the case in Mac. 

Ref.: https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=net-6.0
image

Have added the fix in this [commit](https://github.com/angularsen/UnitsNet/pull/991/commits/3dfe9a4913d25466671355f09ed478cd5d2ae599) together with a small renaming.

**Comment by angularsen** [comment]
Great that you figured it out!

File: UnitsNet.Tests/CustomCode/TemperatureGradientTests.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+{
+    public class TemperatureGradientTests : TemperatureGradientTestsBase
+    {
+        protected override bool SupportsSIUnitSystem => true;
+
+        protected override double DegreesCelciusPerKilometerInOneKelvinPerMeter => 1000;
+
+        protected override double DegreesCelciusPerMeterInOneKelvinPerMeter => 1;
+
+        protected override double DegreesFahrenheitPerFootInOneKelvinPerMeter => 0.54864;
```
Comment (angularsen) [inline UnitsNet.Tests/CustomCode/TemperatureGradientTests.cs:33]: Verified

https://unitconverterapp.com/quantity/temperature%20interval%20per%20length
https://www.calculand.com/unit-converter/?gruppe=Temperature+gradient

File: UnitsNet/UnitsNet.csproj [hunk 2]
```diff
@@ -60,6 +60,9 @@
     <Compile Remove="UnitDefinitions\**" />
     <EmbeddedResource Remove="UnitDefinitions\**" />
     <None Remove="UnitDefinitions\**" />
+    <None Include="..\Common\UnitDefinitions\TemperatureGradient.json">
+      <Link>Common\UnitDefinitions\TemperatureGradient.json</Link>
+    </None>
```
Comment (angularsen) [inline UnitsNet/UnitsNet.csproj]: This was probably accidental, but I can't push to your master branch so I will fix after merging

File: UnitsNet.Tests/CustomCode/TemperatureGradientTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        public void TemperatureDeltaDividedByTemperatureGradientEqualsLength()
+        {
+            Length length = TemperatureDelta.FromDegreesCelsius(50) / TemperatureGradient.FromDegreesCelciusPerKilometer(5);
+            Assert.Equal(length, Length.FromKilometers(10));
+        }
+
+        [Fact]
+        public void TemperatureDeltaDividedByLengthEqualsTemperatureGradient()
+        {
+            TemperatureGradient lapseRate = TemperatureDelta.FromDegreesCelsius(50) / Length.FromKilometers(10);
```
Comment (angularsen) [inline UnitsNet.Tests/CustomCode/TemperatureGradientTests.cs:47]: lapse rate name remains, will fix after merge

**Review by angularsen — APPROVED** [review]
Thanks! Looks good to me, just a few minor things I'll fix after merging.

**Comment by angularsen** [comment]
Nuget should be out shortly.

[Release UnitsNet/4.106.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.106.0)

**Pull request merged**